### ISSUE TITLE
Suggestion: Warn for "using Task" without awaiting it

### ISSUE THREAD

**Issue by markusschaber** [Concurrency-Lab/ParallelHelper#60]
There should be a warning when a Task is directly buried in a using statement, without being awaited. We actually had a production code bug which boils down to something like this:

```
        public interface ICreator
        {
            Task<IDisposable> Create();
        }

        public static async Task Test2Async(ICreator creator)
        {
            using (creator.Create())
            {
                await Console.Out.WriteLineAsync("");
            }
        }
```

But still, there are hardly any cases where tasks really must be disposed, and I cannot imagine a single one where they are disposed without awaiting them (or otherwise making sure that they're finished).

And especially if it's a `Task<IDisposable>`, it's certain that the `Result` of the Task is the one to be `Dispose()`d via `using`.

This could also be generalized to `using` statements, `IAsyncDisposable`, and other awaitables like `ValueTask[<T>]`.

PS:
Of course, it's easier to spot on reviews when the `Create()` method is named `CreateAsync()` instead. We fixed this in the meantime, and established it as a mandatory rule.

**Comment by camrein** [comment]
This is excellent - exactly one of the patterns I long for.  Thank you very much.

**Comment by markusschaber** [comment]
I just tried to test this on our code, and it seems it alerts on Task<IDisposable>, but not Task<IAsyncDisposable>. Could this be added?

**Comment by camrein** [comment]
Thanks for your tests. As you were probably writing your comment, I was already adding the support of `IAsyncDisposable`. I took it a step further: It is not necessary for a type to implement `IAsyncDisposable`, it sufficient to implement an awaitable and parameterless `DisposeAsync` method. Unfortunately, I couldn't find the specification promptly anymore.

PS: I've changed the diagnostic ID to PH_B015 as I think it fits better to the bugs category.
